### PR TITLE
WFCORE-6365 RestartParent OperationStepHandler implementations are in compatible with service built via CapabilityServiceTarget

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFCORE-6365

### CODE REVIEW

**Comment by pferraro** [comment]
Failures in MailSubsystemTestCase are due to https://github.com/wildfly/wildfly/pull/16871

File: controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java [hunk 1]
```diff
@@ -57,19 +57,24 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
...
-                            recreateParentService(context, address, parentModel);
+                            // Remove/recreate parent services in separate step using an OperationContext relative to the parent resource
+                            // This is necessary to support service installation via CapabilityServiceTarget
+                            context.addStep(Util.getReadResourceOperation(address), new OperationStepHandler() {
+                                @Override
+                                public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                                    removeServices(context, serviceName, parentModel);
+                                    recreateParentService(context, parentModel);
+                                }
+                            }, OperationContext.Stage.RUNTIME);
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java:77]: @pferraro I *think* these steps should be added at the top of the queue so they execute prior to any other steps, i.e. in the same relative position as the original code executed them. I pushed up #5528 to get CI going on that while we discuss.

Note that this commit is already merged into main, so please don't update it in this branch; if any change is appropriate it needs to be via a new commit.

**Pull request merged**